src/master/chartsdata.cc

```diff
@@ -169,32 +186,13 @@ void chartsdata_refresh(void) {
 	}
 
 #ifdef CPU_USAGE
-// CPU usage
-	setitimer(ITIMER_VIRTUAL,&it_set,&uc);             // user time
-	setitimer(ITIMER_PROF,&it_set,&pc);                // user time + system time
-
-	if (uc.it_value.tv_sec<=999) {  // on fucken linux timers can go backward !!!
-		uc.it_value.tv_sec = 999-uc.it_value.tv_sec;
-		uc.it_value.tv_usec = 999999-uc.it_value.tv_usec;
+	// CPU usage
+	chartsdata_refresh_cpu_itcp(ITIMER_VIRTUAL, &ucusec); // user time
+	chartsdata_refresh_cpu_itcp(ITIMER_PROF,    &pcusec); // user time + system_time
+	if (pcusec > ucusec) {
+		pcusec -= ucusec;
 	} else {
-		uc.it_value.tv_sec = 0;
-		uc.it_value.tv_usec = 0;
-	}
-	if (pc.it_value.tv_sec<=999) {  // as abowe - who the hell has invented this stupid os !!!
-		pc.it_value.tv_sec = 999-pc.it_value.tv_sec;
-		pc.it_value.tv_usec = 999999-pc.it_value.tv_usec;
-	} else {
-		pc.it_value.tv_sec = 0;
-		uc.it_value.tv_usec = 0;
```

Comment:
It should be noted that this PR modifies the behavoir a bit: the code here seems to be a bit inconsistent, as it modifies both `pc` and `uc`, which didn't seem to be intended.